aspnetcore/blazor/hybrid/tutorials/maui-blazor-web-app.md

```diff
@@ -0,0 +1,322 @@
+
+`InteractiveRenderSettings.cs`:
+
+:::code language="csharp" source="~/../blazor-samples/8.0/MauiBlazorWeb/MauiBlazorWeb.Shared/InteractiveRenderSettings.cs":::
+
+In `MauiProgram.CreateMauiApp` of `MauiProgram.cs`, call `ConfigureBlazorHybridRenderModes`:
+
+```csharp 
+InteractiveRenderSettings.ConfigureBlazorHybridRenderModes();
+```
+
+In the RCL's `_Imports.razor` file, add the following global static `@using` directive to make the properties of the class available to components:
+
+```razor
+@using static InteractiveRenderSettings
+```
+
+At this point, you can adopt either:
+
+* [Global interactive rendering](#global-interactive-rendering)
+* [Per-page/component rendering](#per-pagecomponent-rendering)
+
+### Global interactive rendering
+
+The MAUI project is interactive by default, so no action is taken at the project level in the MAUI project other than calling `InteractiveRenderSettings.ConfigureBlazorHybridRenderModes`.
+
+The Blazor Web App sets a global interactive render mode normally in the `App` component (`Components/App.razor`). In the following example, the `HeadOutlet` and `Routes` components assign `InteractiveServer` from <xref:Microsoft.AspNetCore.Components.Web.RenderMode> via the Blazor Web App's `@using static Microsoft.AspNetCore.Components.Web.RenderMode` directive in the project's `_Imports` file:
+
+```razor
+<HeadOutlet @rendermode="InteractiveServer" />
+
+...
+
+<Routes @rendermode="InteractiveServer" />
+```
+
+At this point, no component in the RCL sets a render mode. A component's render mode is inherited from the global setting.
+
+In summary:
+
+* For components rendered by MAUI for the native client, RCL component render modes are `null`. The components are interactive by default.
+* For components rendered by the Blazor Web App for the web client, RCL component render modes are inherited from the global setting in the `App` component.
+
+### Per-page/component rendering
+
+The MAUI project is interactive by default, so no action is taken at the project level in the MAUI project other than calling `InteractiveRenderSettings.ConfigureBlazorHybridRenderModes`.
+
+When Auto or WebAssembly global interactivity is used, the solution also receives a `.Client` project:
+
+* Add a project reference to the `.Client` project for the RCL.
+* Add an `@using` statement to the `.Client` project's `_Imports.razor` file for the RCL: `@using MauiBlazorWeb.Shared`.
+* Create a second RCL to maintain the Auto or WebAssembly components (for example: `MauiBlazorWeb.Shared.Client`). Move the `Components` folder of the `.Client` project into the second RCL. Create a project reference in the first RCL (`MauiBlazorWeb.Shared`) to the second RCL that maintains the components (`MauiBlazorWeb.Shared.Client`). Add `@using static InteractiveRenderSettings` to the second RCL's `_Imports.razor` file.
```

Comment:
This is ONLY necessary for WebAssembly that also specifies Per page/component rendering. 

So basically it boils down to this -->
When Interactivity Location is Global and Interactivity Platform is Server then:

- There's just three projects (MAUI, Web, Shared)
- No need for the InteractiveRenderSettings class to intercept the render modes in the RCL

When Interactivity Location is Global and Interactivity Platform is Auto or WebAssembly then:

- There's four projects (MAUI, Web, Web.Client, Shared)
- No need for the InteractiveRenderSettings class to intercept the render modes in the RCL

When Interactivity Location is Per page/component and Interactivity Platform is Server then:

- There's just three projects (MAUI, Web, Shared)
- You do need the InteractiveRenderSettings class to intercept the render modes in the RCL and set them to Null in the MauiProgram.cs

When Interactivity Location is Per page/component and Interactivity Platform is Auto then:

- There's four projects (MAUI, Web, Web.Client, Shared)
- You do need the InteractiveRenderSettings class to intercept the render modes in the RCL and set them to Null in the MauiProgram.cs

When Interactivity Location is Per page/component and Interactivity Platform is WebAssembly then:

- There's FIVE projects (MAUI, Web, Web.Client, Shared, Shared.Client)
- The Shared.Client contains all the WebAssembly rendered components and is referenced by the Web.Client (and the Shared RCL)
- You do need the InteractiveRenderSettings class to intercept the render modes in the RCL and set them to Null in the MauiProgram.cs

Reply:
I see ... I'll work on this tomorrow morning.